app/controllers/api/v1/users_controller.rb

```diff
@@ -17,23 +17,28 @@ def show
         render_data data: user, status: :ok
       end
 
+      # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
       # POST /api/v1/users.json
       # Expects: { user: { :name, :email, :password} }
       # Returns: { data: Array[serializable objects] , errors: Array[String] }
       # Does: Creates and saves a new user record in the database with the provided parameters.
 
       def create
-        # TODO: amir - ensure accessibility for unauthenticated requests only.
-        params[:user][:language] = I18n.default_locale if params[:user][:language].blank?
+        # Users created by a user will have the creator language by default with a fallback to the server configured default_locale.
+        user_params[:language] = current_user&.language || I18n.default_locale if user_params[:language].blank?
```

Comment:
This way if the user was created by an admin it will have the admin language by default and not the admin browser language,
A fallback will be the Greenlight server default locale taken from the `.env`,